app/domain/authentication/authn_jwt/functions.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnJwt
+    # Joins "prefix" and "main" as a path, with exactly one slash (/).  If
+    # prefix ends with a slash or host begins with one, they're removed to avoid
+    # duplicate slashes.
+    class << self
+      def path_join(*parts)
+        parts.reduce { |m, x| path_join_two(m, x) }
+      end
+
+      private
+
+      def path_join_two(prefix, main)
```

Comment:
Maybe there is a File library in ruby that has option to join file paths and it can be used to join path here ?

Reply:
File paths depend on operating system though , and most filesystems are forgiving of repetitive path separators, whereas I don't believe Conjur would inherit that same forgiveness when executing database queries or parsing relationships. That said, if there isn't a utility class in the PolicyEngine gem, it may be time to add one starting with this function. I like that it is SRP and much more testable.

Reply:
@tzheleznyak, Ruby has `File.join`, but it is not as flexible as this implementation.  Join does not handle nil or empty strings well and does not strip trailing slashes.

Reply:
@jvanderhoof @tzheleznyak You know, I think you're right about `File.join`.   Even though my implementation handles nils, I think we may want the client code to handle that anyway.  I'm going to try that tomorrow.  If it works, we'll be able to delete my code too!  More code deleted, the better.